flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/enumerator/tracker/SplitTracker.java

```diff
@@ -185,7 +186,8 @@ public List<DynamoDbStreamsShardSplit> getUnassignedChildSplits(Set<String> pare
                             if (!parentChildSplitMap.containsKey(splitId)) {
                                 LOG.warn(
                                         "splitId: {} is not present in parent-child relationship map. "
-                                                + "This indicates that there might be some data loss in the application",
+                                                + "This indicates that either there might be some data loss in the "
+                                                + "application or the child shard hasn't been discovered yet",
```

Comment:
Do we see this very often? If so, this is worrying, and we should create a JIRA to track an improvement. We know our setup is resilient against this, because the child will get picked up eventually, but it would be good to not worry people unnecessarily!

Reply:
yeh, we see this often if we are always at the tip of the log and child shards haven't been discovered yet :)